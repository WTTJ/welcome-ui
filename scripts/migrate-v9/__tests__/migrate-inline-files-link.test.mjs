import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - Link Components', () => {
  const tempDir = setupMigrationTest('inline-link-test', '__fixtures__')

  it('migrates Link components with styling props to Link with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'LinkComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('link-component.tsx')
  })
})
