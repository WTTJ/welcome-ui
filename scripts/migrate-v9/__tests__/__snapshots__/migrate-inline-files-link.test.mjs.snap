// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Link Components > migrates Link components with FormattedMessage to a with classNames > link-component.tsx 1`] = `
"import { Button } from 'welcome-ui/Button'
import { Link } from 'welcome-ui/Link'

const FormattedMessage = ({
  defaultMessage,
  id,
}: {
  defaultMessage: string
  id: string
}): React.ReactNode => <span id={id}>{defaultMessage}</span>

const SIGN_IN_PATH = '/signin'

export const LinkComponent = () => {
  return (
    <div>
      <Link as={Button} to={SIGN_IN_PATH}>
        <FormattedMessage defaultMessage="Terms of use" id="EBDrtq" />
      </Link>
    </div>
  )
}
"
`;
